agent/lib/kontena/workers/log_worker.rb

```diff
@@ -166,21 +172,23 @@ def stop_streaming
       info 'stop log streaming'
 
       @workers.keys.dup.each do |id|
+        queued_item = @queue.find { |i| i[:id] == id }
+        time = queued_item.nil? ? Time.now.to_i : Time.parse(queued_item[:time]).to_i
```

Comment:
The use of `to_i` leads to duplicate logs across an agent restart, but that's already an issue:

```
2017-09-05T11:40:37.000Z [1]: 2017-09-05 11:40:37.821891553+00:00
2017-09-05T11:40:37.000Z [1]: 2017-09-05 11:40:37.923126365+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.024403013+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.129171527+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.232594795+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.334944198+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.436267847+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.539069666+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.640869351+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.742146178+00:00

2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.024403013+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.129171527+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.232594795+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.334944198+00:00
2017-09-05T11:40:38.000Z [1]: 2017-09-05 11:40:38.436267847+00:00
```